### ISSUE TITLE
Heroku Website Error(new)

### ISSUE THREAD

**Issue by sadmanahmed** [ormprog/rails_ama#131]
```
abir@abir:~/Ruby code/blog$ heroku logs -t -a abir-blog-1
 ›   Warning: heroku update available from 7.46.2 to 7.47.0.
2020-11-10T13:27:24.224864+00:00 app[web.1]: => Run `rails server --help` for more startup options
2020-11-10T13:27:25.825558+00:00 app[web.1]: Puma starting in single mode...
2020-11-10T13:27:25.825682+00:00 app[web.1]: * Version 4.3.6 (ruby 2.6.5-p114), codename: Mysterious Traveller
2020-11-10T13:27:25.825727+00:00 app[web.1]: * Min threads: 5, max threads: 5
2020-11-10T13:27:25.825817+00:00 app[web.1]: * Environment: production
2020-11-10T13:27:25.826347+00:00 app[web.1]: * Listening on tcp://0.0.0.0:58978
2020-11-10T13:27:25.826913+00:00 app[web.1]: Use Ctrl-C to stop
2020-11-10T13:27:26.381336+00:00 heroku[web.1]: State changed from starting to up
2020-11-10T13:28:15.330645+00:00 app[api]: Starting process with command `rails db:migrate` by user abirsadmanahmed177@gmail.com
2020-11-10T13:28:49.401389+00:00 heroku[run.7660]: State changed from starting to up
2020-11-10T13:28:49.789417+00:00 heroku[run.7660]: Awaiting client
2020-11-10T13:29:19.825598+00:00 heroku[run.7660]: Process exited with status 128
2020-11-10T13:29:19.826441+00:00 heroku[run.7660]: Error R13 (Attach error) -> Failed to attach to process
2020-11-10T13:29:19.873107+00:00 heroku[run.7660]: State changed from up to complete
2020-11-10T13:59:09.984340+00:00 app[api]: Starting process with command `rails db:migrate` by user abirsadmanahmed177@gmail.com
2020-11-10T13:59:23.846616+00:00 heroku[run.1638]: Idling
2020-11-10T13:59:23.849209+00:00 heroku[run.1638]: State changed from starting to complete
2020-11-10T13:59:23.854717+00:00 heroku[web.1]: Idling
2020-11-10T13:59:23.857032+00:00 heroku[web.1]: State changed from up to down
2020-11-10T13:59:25.098309+00:00 heroku[web.1]: Stopping all processes with SIGTERM
2020-11-10T13:59:25.155138+00:00 app[web.1]: - Gracefully stopping, waiting for requests to finish
2020-11-10T13:59:25.168322+00:00 app[web.1]: === puma shutdown: 2020-11-10 13:59:25 +0000 ===
2020-11-10T13:59:25.168378+00:00 app[web.1]: - Goodbye!
2020-11-10T13:59:25.168622+00:00 app[web.1]: Exiting
2020-11-10T13:59:25.279891+00:00 heroku[web.1]: Process exited with status 143
2020-11-10T13:59:38.402206+00:00 heroku[run.1638]: Awaiting client
2020-11-10T13:59:38.423670+00:00 heroku[run.1638]: Starting process with command `rails db:migrate`
2020-11-10T14:00:19.036185+00:00 heroku[web.1]: Unidling
2020-11-10T14:00:19.038262+00:00 heroku[web.1]: State changed from down to starting
2020-11-10T14:00:36.154446+00:00 heroku[web.1]: Starting process with command `bin/rails server -p ${PORT:-5000} -e production`
2020-11-10T14:00:45.422377+00:00 app[web.1]: => Booting Puma
2020-11-10T14:00:45.422390+00:00 app[web.1]: => Rails 6.0.3.4 application starting in production
2020-11-10T14:00:45.422391+00:00 app[web.1]: => Run `rails server --help` for more startup options
2020-11-10T14:00:47.451653+00:00 app[web.1]: Puma starting in single mode...
2020-11-10T14:00:47.451669+00:00 app[web.1]: * Version 4.3.6 (ruby 2.6.5-p114), codename: Mysterious Traveller
2020-11-10T14:00:47.451670+00:00 app[web.1]: * Min threads: 5, max threads: 5
2020-11-10T14:00:47.451703+00:00 app[web.1]: * Environment: production
2020-11-10T14:00:47.452111+00:00 app[web.1]: * Listening on tcp://0.0.0.0:39733
2020-11-10T14:00:47.452898+00:00 app[web.1]: Use Ctrl-C to stop
2020-11-10T14:00:48.026840+00:00 heroku[web.1]: State changed from starting to up
2020-11-10T14:00:49.131157+00:00 app[web.1]: I, [2020-11-10T14:00:49.131014 #4]  INFO -- : [5b8e9918-ce45-455c-9fef-42b192f0df3c] Started GET "/" for 103.125.235.22 at 2020-11-10 14:00:49 +0000
2020-11-10T14:00:49.134298+00:00 app[web.1]: I, [2020-11-10T14:00:49.134199 #4]  INFO -- : [5b8e9918-ce45-455c-9fef-42b192f0df3c] Processing by PostsController#index as HTML
2020-11-10T14:00:49.203479+00:00 app[web.1]: I, [2020-11-10T14:00:49.203349 #4]  INFO -- : [5b8e9918-ce45-455c-9fef-42b192f0df3c] Completed 500 Internal Server Error in 69ms (ActiveRecord: 25.1ms | Allocations: 3931)
2020-11-10T14:00:49.204212+00:00 app[web.1]: F, [2020-11-10T14:00:49.204128 #4] FATAL -- : [5b8e9918-ce45-455c-9fef-42b192f0df3c]
2020-11-10T14:00:49.204213+00:00 app[web.1]: [5b8e9918-ce45-455c-9fef-42b192f0df3c] ActiveRecord::StatementInvalid (PG::UndefinedTable: ERROR:  relation "posts" does not exist
2020-11-10T14:00:49.204214+00:00 app[web.1]: LINE 8:  WHERE a.attrelid = '"posts"'::regclass
2020-11-10T14:00:49.204215+00:00 app[web.1]: ^
2020-11-10T14:00:49.204216+00:00 app[web.1]: ):
2020-11-10T14:00:49.204216+00:00 app[web.1]: [5b8e9918-ce45-455c-9fef-42b192f0df3c]
2020-11-10T14:00:49.204217+00:00 app[web.1]: [5b8e9918-ce45-455c-9fef-42b192f0df3c] app/controllers/posts_controller.rb:6:in `index'
2020-11-10T14:00:49.205764+00:00 heroku[router]: at=info method=GET path="/" host=abir-blog-1.herokuapp.com request_id=5b8e9918-ce45-455c-9fef-42b192f0df3c fwd="103.125.235.22" dyno=web.1 connect=0ms service=76ms status=500 bytes=1827 protocol=https
2020-11-10T14:00:49.732128+00:00 heroku[router]: at=info method=GET path="/favicon.ico" host=abir-blog-1.herokuapp.com request_id=ec3f48e4-9bd4-4b47-b5bf-938376eabddd fwd="103.125.235.22" dyno=web.1 connect=0ms service=17ms status=200 bytes=143 protocol=https
2020-11-10T14:00:53.065525+00:00 app[web.1]: I, [2020-11-10T14:00:53.065388 #4]  INFO -- : [872b6831-c0d3-40bd-ba7b-196738288819] Started GET "/" for 103.125.235.22 at 2020-11-10 14:00:53 +0000
2020-11-10T14:00:53.066488+00:00 app[web.1]: I, [2020-11-10T14:00:53.066406 #4]  INFO -- : [872b6831-c0d3-40bd-ba7b-196738288819] Processing by PostsController#index as HTML
2020-11-10T14:00:53.078572+00:00 app[web.1]: I, [2020-11-10T14:00:53.078465 #4]  INFO -- : [872b6831-c0d3-40bd-ba7b-196738288819] Completed 500 Internal Server Error in 12ms (ActiveRecord: 6.6ms | Allocations: 325)
2020-11-10T14:00:53.079816+00:00 app[web.1]: F, [2020-11-10T14:00:53.079718 #4] FATAL -- : [872b6831-c0d3-40bd-ba7b-196738288819]
2020-11-10T14:00:53.079817+00:00 app[web.1]: [872b6831-c0d3-40bd-ba7b-196738288819] ActiveRecord::StatementInvalid (PG::UndefinedTable: ERROR:  relation "posts" does not exist
2020-11-10T14:00:53.079818+00:00 app[web.1]: LINE 8:  WHERE a.attrelid = '"posts"'::regclass
2020-11-10T14:00:53.079819+00:00 app[web.1]: ^
2020-11-10T14:00:53.079819+00:00 app[web.1]: ):
2020-11-10T14:00:53.079820+00:00 app[web.1]: [872b6831-c0d3-40bd-ba7b-196738288819]
2020-11-10T14:00:53.079820+00:00 app[web.1]: [872b6831-c0d3-40bd-ba7b-196738288819] app/controllers/posts_controller.rb:6:in `index'
2020-11-10T14:00:53.081296+00:00 heroku[router]: at=info method=GET path="/" host=abir-blog-1.herokuapp.com request_id=872b6831-c0d3-40bd-ba7b-196738288819 fwd="103.125.235.22" dyno=web.1 connect=0ms service=16ms status=500 bytes=1827 protocol=https
2020-11-10T14:00:53.562286+00:00 heroku[router]: at=info method=GET path="/favicon.ico" host=abir-blog-1.herokuapp.com request_id=f8a13b22-91f8-47bb-872f-68127f409e4f fwd="103.125.235.22" dyno=web.1 connect=0ms service=1ms status=200 bytes=143 protocol=https
2020-11-10T14:01:07.070732+00:00 app[web.1]: I, [2020-11-10T14:01:07.070629 #4]  INFO -- : [9bd97f89-9df0-4a64-a8c6-e06d562ba9c5] Started GET "/" for 116.204.255.238 at 2020-11-10 14:01:07 +0000
2020-11-10T14:01:07.071501+00:00 app[web.1]: I, [2020-11-10T14:01:07.071415 #4]  INFO -- : [9bd97f89-9df0-4a64-a8c6-e06d562ba9c5] Processing by PostsController#index as HTML
2020-11-10T14:01:07.076801+00:00 app[web.1]: I, [2020-11-10T14:01:07.076694 #4]  INFO -- : [9bd97f89-9df0-4a64-a8c6-e06d562ba9c5] Completed 500 Internal Server Error in 5ms (ActiveRecord: 2.3ms | Allocations: 325)
2020-11-10T14:01:07.078320+00:00 app[web.1]: F, [2020-11-10T14:01:07.078211 #4] FATAL -- : [9bd97f89-9df0-4a64-a8c6-e06d562ba9c5]
2020-11-10T14:01:07.078321+00:00 app[web.1]: [9bd97f89-9df0-4a64-a8c6-e06d562ba9c5] ActiveRecord::StatementInvalid (PG::UndefinedTable: ERROR:  relation "posts" does not exist
2020-11-10T14:01:07.078321+00:00 app[web.1]: LINE 8:  WHERE a.attrelid = '"posts"'::regclass
2020-11-10T14:01:07.078322+00:00 app[web.1]: ^
2020-11-10T14:01:07.078322+00:00 app[web.1]: ):
2020-11-10T14:01:07.078322+00:00 app[web.1]: [9bd97f89-9df0-4a64-a8c6-e06d562ba9c5]
2020-11-10T14:01:07.078323+00:00 app[web.1]: [9bd97f89-9df0-4a64-a8c6-e06d562ba9c5] app/controllers/posts_controller.rb:6:in `index'
2020-11-10T14:01:07.080769+00:00 heroku[router]: at=info method=GET path="/" host=abir-blog-1.herokuapp.com request_id=9bd97f89-9df0-4a64-a8c6-e06d562ba9c5 fwd="116.204.255.238" dyno=web.1 connect=1ms service=10ms status=500 bytes=1827 protocol=https
2020-11-10T14:01:07.999416+00:00 heroku[router]: at=info method=GET path="/favicon.ico" host=abir-blog-1.herokuapp.com request_id=53265f43-5dee-4ce2-a232-26622c60fe38 fwd="116.204.255.238" dyno=web.1 connect=1ms service=7ms status=200 bytes=143 protocol=https
2020-11-10T14:01:09.549065+00:00 app[web.1]: I, [2020-11-10T14:01:09.548917 #4]  INFO -- : [00093a1e-4027-450e-b022-239f27100d20] Started GET "/" for 116.204.255.238 at 2020-11-10 14:01:09 +0000
2020-11-10T14:01:09.549931+00:00 app[web.1]: I, [2020-11-10T14:01:09.549849 #4]  INFO -- : [00093a1e-4027-450e-b022-239f27100d20] Processing by PostsController#index as HTML
2020-11-10T14:01:09.557162+00:00 app[web.1]: I, [2020-11-10T14:01:09.557090 #4]  INFO -- : [00093a1e-4027-450e-b022-239f27100d20] Completed 500 Internal Server Error in 7ms (ActiveRecord: 3.0ms | Allocations: 325)
2020-11-10T14:01:09.558352+00:00 app[web.1]: F, [2020-11-10T14:01:09.558290 #4] FATAL -- : [00093a1e-4027-450e-b022-239f27100d20]
2020-11-10T14:01:09.558353+00:00 app[web.1]: [00093a1e-4027-450e-b022-239f27100d20] ActiveRecord::StatementInvalid (PG::UndefinedTable: ERROR:  relation "posts" does not exist
2020-11-10T14:01:09.558354+00:00 app[web.1]: LINE 8:  WHERE a.attrelid = '"posts"'::regclass
2020-11-10T14:01:09.558355+00:00 app[web.1]: ^
2020-11-10T14:01:09.558355+00:00 app[web.1]: ):
2020-11-10T14:01:09.558356+00:00 app[web.1]: [00093a1e-4027-450e-b022-239f27100d20]
2020-11-10T14:01:09.558356+00:00 app[web.1]: [00093a1e-4027-450e-b022-239f27100d20] app/controllers/posts_controller.rb:6:in `index'
2020-11-10T14:01:09.560031+00:00 heroku[router]: at=info method=GET path="/" host=abir-blog-1.herokuapp.com request_id=00093a1e-4027-450e-b022-239f27100d20 fwd="116.204.255.238" dyno=web.1 connect=1ms service=13ms status=500 bytes=1827 protocol=https
2020-11-10T14:01:10.032609+00:00 heroku[router]: at=info method=GET path="/favicon.ico" host=abir-blog-1.herokuapp.com request_id=d717fdf8-7cae-410a-a3ab-f322cd498b9d fwd="116.204.255.238" dyno=web.1 connect=1ms service=2ms status=200 bytes=143 protocol=https
2020-11-10T14:01:10.528016+00:00 app[web.1]: I, [2020-11-10T14:01:10.527830 #4]  INFO -- : [7d175eda-490c-4679-a0b4-e3884f1e0405] Started GET "/" for 116.204.255.238 at 2020-11-10 14:01:10 +0000
2020-11-10T14:01:10.528893+00:00 app[web.1]: I, [2020-11-10T14:01:10.528823 #4]  INFO -- : [7d175eda-490c-4679-a0b4-e3884f1e0405] Processing by PostsController#index as HTML
2020-11-10T14:01:10.537267+00:00 app[web.1]: I, [2020-11-10T14:01:10.537172 #4]  INFO -- : [7d175eda-490c-4679-a0b4-e3884f1e0405] Completed 500 Internal Server Error in 5ms (ActiveRecord: 2.2ms | Allocations: 325)
2020-11-10T14:01:10.538457+00:00 app[web.1]: F, [2020-11-10T14:01:10.538388 #4] FATAL -- : [7d175eda-490c-4679-a0b4-e3884f1e0405]
2020-11-10T14:01:10.538458+00:00 app[web.1]: [7d175eda-490c-4679-a0b4-e3884f1e0405] ActiveRecord::StatementInvalid (PG::UndefinedTable: ERROR:  relation "posts" does not exist
2020-11-10T14:01:10.538459+00:00 app[web.1]: LINE 8:  WHERE a.attrelid = '"posts"'::regclass
2020-11-10T14:01:10.538460+00:00 app[web.1]: ^
2020-11-10T14:01:10.538460+00:00 app[web.1]: ):
2020-11-10T14:01:10.538460+00:00 app[web.1]: [7d175eda-490c-4679-a0b4-e3884f1e0405]
2020-11-10T14:01:10.538461+00:00 app[web.1]: [7d175eda-490c-4679-a0b4-e3884f1e0405] app/controllers/posts_controller.rb:6:in `index'
2020-11-10T14:01:10.540996+00:00 heroku[router]: at=info method=GET path="/" host=abir-blog-1.herokuapp.com request_id=7d175eda-490c-4679-a0b4-e3884f1e0405 fwd="116.204.255.238" dyno=web.1 connect=1ms service=16ms status=500 bytes=1827 protocol=https
2020-11-10T14:01:11.014816+00:00 heroku[router]: at=info method=GET path="/favicon.ico" host=abir-blog-1.herokuapp.com request_id=bcc3786f-ff78-41a9-a668-b25947d892e5 fwd="116.204.255.238" dyno=web.1 connect=5ms service=4ms status=200 bytes=143 protocol=https

```

I can see this error `(PG::UndefinedTable: ERROR:  relation "posts" does not exist`

**Comment by wasifhossain** [comment]
i guess you missed to apply the migration

**Comment by sadmanahmed** [comment]
I did it already before facing the error 
```
abir@abir:~/Ruby code/blog$ heroku run rails db:migrate
 ›   Warning: heroku update available from 7.46.2 to 7.47.0.
Running rails db:migrate on ⬢ abir-blog-1... up, run.1638 (Free)
```

**Comment by sadmanahmed** [comment]
okay ,again migration fixed the issue
